lib/pipelines/code-pipeline.ts

```diff
@@ -255,19 +290,35 @@ export class ApplicationStage extends cdk.Stage {
 class CodePipeline {
 
     public static build(scope: Construct, props: PipelineProps) : cdkpipelines.CodePipeline {
-        const branch = props.repository.targetRevision ?? 'main';
-        let githubProps : cdkpipelines.GitHubSourceOptions | undefined = undefined;
-
-        if(props.repository.credentialsSecretName) {
-            githubProps = {
-                authentication: cdk.SecretValue.secretsManager(props.repository.credentialsSecretName!)
-            };
+        let codePipelineSource : cdkpipelines.CodePipelineSource | undefined = undefined;
+
+        if (isCodeCommitRepo(props.repository)) {
+            let codeCommitRepo = props.repository as CodeCommitSourceRepository;
+            codePipelineSource = cdkpipelines.CodePipelineSource.codeCommit(
+                codecommit.Repository.fromRepositoryName(
+                    scope, 'cdk-eks-blueprints', codeCommitRepo.codeCommitRepoName),
+                    codeCommitRepo.targetRevision ?? 'master',
+                    codeCommitRepo.codeCommitOptions);
+        } else {
+            let gitHubRepo = props.repository as GitHubSourceRepository;
+            let githubProps : cdkpipelines.GitHubSourceOptions | undefined = undefined;
+            let gitHubOwner = gitHubRepo.owner ? gitHubRepo.owner : gitHubRepo.owner;
```

Comment:
@ybezsonov this looks like a defect: I think you meant to use `gitHubRepo.owner` if defined and fall back to  `props.owner` otherwise. Otherwise, it will be a breaking change.

Please consider nullish coalscing for string defaulting:

```typescript
const gitHubOwner = gitHubRepo.owner ?? props.owner;
```

Reply:
It was copy/paste error, but quite dangerous one.
Thank you for spotting this!